packages/web/src/dev/ValidatorsList.tsx

```diff
@@ -58,261 +54,78 @@ class HeaderCell extends React.PureComponent<HeaderCellProps, { hover: boolean }
-            name: group.name,
-            address: group.address,
-            usd: weiToDecimal(+group.usd),
-            gold: weiToDecimal(+group.lockedGold),
-            receivableRaw: weiToDecimal(+receivableVotes),
-            receivableVotes: receivableVotesPer.toString(),
-            votesRaw: weiToDecimal(+votes),
-            votes: votesPer.toString(),
-            votesAbsolute: votesAbsolutePer.toString(),
-            commission: (+commission * 100) / 10 ** 24,
-            rewards,
-            rewardsStyle,
-            numMembers,
-            claims: getClaims(group.claims),
-            validators: affiliates.edges.map(({ node: validator }) => {
-              const {
-                address,
-                lastElected,
-                lastOnline,
-                name,
-                usd,
-                lockedGold,
-                score,
-                attestationsFulfilled,
-                attestationsRequested,
-              } = validator
-              return {
-                name,
-                address,
-                usd: weiToDecimal(+usd),
-                gold: weiToDecimal(+lockedGold),
-                elected: lastElected >= latestBlock,
-                online: lastOnline >= latestBlock,
-                uptime: (+score * 100) / 10 ** 24,
-                attestation:
-                  Math.max(0, attestationsFulfilled / (attestationsRequested || -1)) * 100,
-                claims: getClaims(validator.account.claims),
-              }
-            }),
-          }
-        }
-      )
-      .map((group, id) => {
-        const data = group.validators.reduce(
-          ({ elected, online, total, uptime }, validator) => ({
-            elected: elected + +validator.elected,
-            online: online + +validator.online,
-            total: total + 1,
-            uptime: uptime + validator.uptime,
-          }),
-          { elected: 0, online: 0, total: 0, uptime: 0 }
-        )
-        data.uptime = data.uptime / group.validators.length
-        return {
-          id,
-          ...group,
-          ...data,
-        }
-      })
-    this.cachedCleanData = cleanData
-    return cleanData
+  setData = () => {
+    const { orderAsc, orderKey } = this.state
+    const sortedData = sortData(this.props.data, orderAsc, orderKey)
+    this.setState({ sortedData })
+    this.forceUpdate()
   }
 
-  sortData<T extends any & { id: number }>(data: T[]): T[] {
-    const { orderBy, orderAsc } = this.state
-    const accessor = this.orderAccessors[orderBy] || (() => 0)
-    const dAccessor = this.orderAccessors[this.defaultOrderAccessor]
-    const dir = orderAsc ? 1 : -1
-
-    const compare = (a, b): number => {
-      if (a === null) {
-        return 1
-      }
-      if (b === null) {
-        return -1
-      }
-      return a > b ? 1 : -1
+  expand(expanded: number) {
+    if (this.state.expanded === expanded) {
+      return this.setState({ expanded: undefined })
```

Comment:
oh! was about to write that it looked like both setStates were being called since i didnt see this return statement .